exp/internal/webhooks/machinepool.go

```diff
@@ -108,6 +112,11 @@ func (webhook *MachinePool) Default(ctx context.Context, obj runtime.Object) err
 		m.Spec.Template.Spec.InfrastructureRef.Namespace = m.Namespace
 	}
 
+	// Set the default value for the node deletion timeout.
+	if m.Spec.Template.Spec.NodeDeletionTimeout == nil {
+		m.Spec.Template.Spec.NodeDeletionTimeout = &metav1.Duration{Duration: defaultNodeDeletionTimeout}
+	}
```

Comment:
Adding defaulting to 10 seconds IMHO is a change in the behaviour of the controller.

Should we mark this as feature (❇️ ) or breaking change in the tilte (✨ )?

Note: this should not get cherry-picked to older release branches.